File: commit_counter.py
#
# usage:
#   python commit_counter.py username [since]
#   - username: github username
#   - since: optional date value, 2019-08-05T00:00:00Z will be used if not present
#            date format ISO 8601: YYYY-MM-DDTHH:MM:SSZ
#
import requests
import sys
import json

base = 'https://api.github.com'
user = sys.argv[1]
since = '2019-08-05T00:00:00Z'
if len(sys.argv) > 2:
    since = sys.argv[2]


resp = requests.get(base + '/users/' +  user + '/repos')

if resp.status_code == 200:
    repos = json.loads(resp.content)
    for repo in repos:
        print(repo['full_name'])
        commits_resp = requests.get(base + '/repos/' + repo['full_name'] + '/commits?author=' + user + '&since=' + since)
        if commits_resp.status_code == 200:
            commits = json.loads(commits_resp.content)
            print(len(commits))

File: commit_counter.cpp
#define CPPHTTPLIB_OPENSSL_SUPPORT
#include "yhirose/httplib.h"    // https://github.com/yhirose/cpp-httplib
#include "nlohmann/json.hpp"    // https://github.com/nlohmann/json

#include <iostream>

using json = nlohmann::json;

int main(int argc, char **argv)
{
    httplib::SSLClient cli("api.github.com");
    std::string user = argv[1];
    std::string since = argc == 3 ? argv[2] : "2019-08-05T00:00:00Z";
    auto res = cli.Get(std::string("/users/" + user + "/repos").c_str());
    if (res && res->status == 200)
    {
        auto repos = json::parse(res->body);
        for (auto& repo : repos)
        {
            std::string repo_name = repo["full_name"];
            std::cout << repo_name << "\n";
            auto commits_res = cli.Get(
                std::string("/repos/" + repo_name + "/commits?author=" + user + "&since=" + since).c_str());
            if (commits_res && commits_res->status == 200)
            {
                auto commits_json = json::parse(commits_res->body);
                std::cout << commits_json.size() << "\n";
            }
        }
    }
}
